src/sql/src/plan/statement/ddl.rs

```diff
@@ -3663,16 +3663,58 @@ pub fn describe_alter_source(
     _: &StatementContext,
     _: AlterSourceStatement<Aug>,
 ) -> Result<StatementDesc, PlanError> {
+    // TODO: put the options here, right?
     Ok(StatementDesc::new(None))
 }
 
 pub fn plan_alter_source(
-    scx: &StatementContext,
+    scx: &mut StatementContext,
     stmt: AlterSourceStatement<Aug>,
 ) -> Result<Plan, PlanError> {
     scx.require_unsafe_mode("ALTER SOURCE")?;
-    let _: AlterSourceAction<Aug> = stmt.action;
-    let _: AlterSourcePlan = sql_bail!("ALTER SOURCE not yet implemented!");
+    let AlterSourceStatement {
+        source_name,
+        if_exists,
+        action,
+    } = stmt;
+    let source_name = normalize::unresolved_object_name(source_name)?;
+    let entry = match scx.catalog.resolve_item(&source_name) {
+        Ok(source) => source,
+        Err(_) if if_exists => {
+            return Ok(Plan::AlterNoop(AlterNoopPlan {
+                object_type: ObjectType::Source,
+            }));
+        }
+        Err(e) => return Err(e.into()),
+    };
+    if entry.item_type() != CatalogItemType::Source {
+        sql_bail!(
+            "{} is a {} not a source",
+            scx.catalog.resolve_full_name(entry.name()),
+            entry.item_type()
+        )
+    }
+    let id = entry.id();
+
+    let config = match action {
+        AlterSourceAction::SetOptions(options) => {
+            let CreateSourceOptionExtracted { size, remote, .. } =
+                CreateSourceOptionExtracted::try_from(options)?;
+
+            match (remote, size) {
+                (None, None) => None,
+                (None, Some(size)) => Some(StorageHostConfig::Managed { size }),
+                (Some(addr), None) => Some(StorageHostConfig::Remote { addr }),
+                (Some(_), Some(_)) => sql_bail!("only one of REMOTE and SIZE can be set"),
+            }
+        }
+        AlterSourceAction::ResetOptions(_) => {
+            // TODO: decide what do do about resetting eg. REMOTE when we're currently managed
+            Some(StorageHostConfig::Undefined)
```

Comment:
Hm, this looks very strange to me, that we have an `Undefined` variant as the output of planning because in my mind planning is where we decide what needs to happen so if a user didn't specify a size I would expect us to select the default size during planning and commit that to the catalog. It looks like an environmentd restart could cause us to reschedule a bunch of sources if the default ever changed.

This isn't something to fix in this PR I just wanted to see if that makes sense to you too

Reply:
though iiuc it's the case that, if no size is specified at `CREATE SOURCE` time, the catalog's stash will not specify a specific size. (since it only persists the create_sql statement, effectively.) that particular issue does not affect the `ALTER SOURCE` path, since the modified create statement we add to the stash always has one or the other option set.

worth opening a ticket for?